File: src/main/java/com/codecool/repvault/domain/utils/FriendRequestUtil.java
package com.codecool.repvault.domain.utils;

import com.codecool.repvault.domain.entities.User;
import com.codecool.repvault.domain.entities.ids.FriendRequestId;
import com.codecool.repvault.infrastructure.repositories.FriendRequestRepository;
import org.springframework.stereotype.Component;

@Component
public class FriendRequestUtil {
    private final SecurityUtil securityUtil;
    private final FriendRequestRepository friendRequestRepository;

    public FriendRequestUtil(SecurityUtil securityUtil, FriendRequestRepository friendRequestRepository) {
        this.securityUtil = securityUtil;
        this.friendRequestRepository = friendRequestRepository;
    }

    public boolean hasRequestWith(User user) {
        User authenticatedUser = securityUtil.getAuthenticatedUser();
        return friendRequestRepository.existsBetween(user.getId(), authenticatedUser.getId());
    }

    public boolean hasRequestFrom(User user) {
        User authenticatedUser = securityUtil.getAuthenticatedUser();
        FriendRequestId id = new FriendRequestId(user.getId(), authenticatedUser.getId());
        return friendRequestRepository.existsById(id);
    }

    public boolean hasRequestTo(User user) {
        User authenticatedUser = securityUtil.getAuthenticatedUser();
        FriendRequestId id = new FriendRequestId(authenticatedUser.getId(), user.getId());
        return friendRequestRepository.existsById(id);
    }
}

File: src/main/java/com/codecool/repvault/domain/utils/FriendRequestUtil.kt
package com.codecool.repvault.domain.utils

import com.codecool.repvault.domain.entities.User
import com.codecool.repvault.domain.entities.ids.FriendRequestId
import com.codecool.repvault.infrastructure.repositories.FriendRequestRepository
import org.springframework.stereotype.Component

@Component
class FriendRequestUtil(
    private val securityUtil: SecurityUtil,
    private val friendRequestRepository: FriendRequestRepository
) {
    fun hasRequestWith(user: User): Boolean {
        val authenticatedUser = securityUtil.authenticatedUser
        return friendRequestRepository.existsBetween(user.id!!, authenticatedUser.id!!)
    }

    fun hasRequestFrom(user: User): Boolean {
        val authenticatedUser = securityUtil.authenticatedUser
        val id = FriendRequestId(user.id!!, authenticatedUser.id!!)
        return friendRequestRepository.existsById(id)
    }

    fun hasRequestTo(user: User): Boolean {
        val authenticatedUser = securityUtil.authenticatedUser
        val id = FriendRequestId(authenticatedUser.id!!, user.id!!)
        return friendRequestRepository.existsById(id)
    }
}
